### ISSUE TITLE
Changing list content length but not updating number of pages

### ISSUE THREAD

**Issue by christraverse** [beneverard/jqPagination#45]
I'm paginating some data from a json file and when I load the page it works fine.  I've written a search function for the data that leads to the number of items in the array being reduced (in this case, down to 5 pages of results), yet the 'first' and 'last' buttons when pressed revert to being of 11 pages, and in the case of page 11 where no records exist displays a blank page.

I don't know if I'm explaining this fully, but if you go to http://jctrl-dev.azurewebsites.net/pickinfo/ and use the search term 'collect' and then use the first page or last page buttons you'll see what I mean.

**Comment by echoate** [comment]
I had the exact same issue.  After stepping through the code, I noticed my page handler was getting called twice when I clicked on the pagination links.  Turns out when you call  $('.pagination').jqPagination() multiple times (i.e. when you call it again after initial page load) it binds another instance of your page handler to the onclick events.  Now both instances of your page handler get called when you click on the pagination links, causing all kinds of unpredictable results as you noticed.  This jQuery behavior is documented here:

http://api.jquery.com/on/

"As of jQuery 1.4, the same event handler can be bound to an element multiple times."

An apparent solution is described here:

http://www.prozblog.com/2012/03/05/jquery-always-use-unbind-or-off-method-to-prevent-event-bubbling/

In other words, in jquery.jqpagination.js change:

         base.$el.on('click.jqPagination', 'a', function (event) {

to:

         base.$el.off('click.jqPagination').on('click.jqPagination', 'a', function (event) {

which results in subsequent initializations of jqPagination after initial page load to work as expected.

Good luck!
